Gemfile

```diff
@@ -1,22 +1,25 @@
 source 'https://rubygems.org'
 gemspec
 
+tilt_version = ENV['TILT_VERSION']
+rack_version = ENV['RACK_VERSION']
+
 # Stick with older racc until
 # https://github.com/tenderlove/racc/issues/22
 # is solved.
 gem 'racc', '< 1.4.10' if RUBY_ENGINE == 'jruby'
 gem 'json', '< 1.8.1'  if RUBY_VERSION.to_f == 2.1 and RUBY_ENGINE == 'ruby'
 gem 'rubysl', :platform => :rbx
-gem 'thin', platform: :mri
+gem 'thin', platform: :mri unless rack_version
```

Comment:
Why is this needed?

If it can't be avoided an explanatory comment will probably help future readers (e.g. me in 5min :stuck_out_tongue_winking_eye:)